### PR TITLE
entitlement tests

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description
Added unit tests for entitlements directory components. Besides the addition of unit tests, disabled check in the entitlement forms has been added. The previous check was only adding the disabled class but not affecting the behavior.

Coverage: 64%

### CODE REVIEW

File: src/users/entitlements/CreateEntitlementForm.test.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    it('Submit button disabled by default', () => {
+      expect(wrapper.find('button.btn-primary').prop('disabled')).toBeTruthy();
+    });
+
+    it('Successful form submission', async () => {
+      const apiMock = jest.spyOn(api, 'postEntitlement').mockImplementationOnce(() => Promise.resolve({}));
+      expect(apiMock).toHaveBeenCalledTimes(0);
+
+      wrapper.find('textarea#comments').simulate('change', { target: { value: 'creating new entitlement' } });
+      const submitButton = wrapper.find('button.btn-primary');
```
Comment (awaisdar001) [inline src/users/entitlements/CreateEntitlementForm.test.jsx:47]: initialization of `submitButton` should go in the top block.

**Review by awaisdar001 — APPROVED** [review]
Looks good to me 👍

**Review by Ali-D-Akbar — APPROVED** [review]
LGTM.

File: src/users/entitlements/CreateEntitlementForm.test.jsx (hunk 1 above)
Comment (DawoudSheraz) [reply src/users/entitlements/CreateEntitlementForm.test.jsx:47]: actually, this causes an error. Once the comments are added, only then the submit button state updates. I can move the initialization above but this line will still remain here.

Comment (awaisdar001) [reply src/users/entitlements/CreateEntitlementForm.test.jsx:47]: got it. makes sense.

**Pull request merged**